### ISSUE TITLE
Providing Date formats and date adapters doesnt work Angular5 material

### ISSUE THREAD

**Issue by CurtisDS** [stackblitz/core#279]
On the Angular website it gives an example of how to change the format of the date picker. 

https://material.angular.io/components/datepicker/overview#customizing-the-parse-and-display-formats

This does not work in stackblitz. The provided date formats are ignored, the initial date value is not set, and the provided date adapter seems to be ignored too.

The example works in my local implementation but not in stackblitz.

**Comment by seveves** [comment]
May you provide a stackblitz example?

**Comment by random11x** [comment]
Right from angular site: https://stackblitz.com/angular/revqynyravm?file=app%2Fdatepicker-formats-example.ts

Date format is ignored.

**Comment by seveves** [comment]
I tried to use momentjs directly via unpkg.com but that didn't help. I'm not sure whats the problem but I bet it has something to do with momentjs. Does it work when running the exact same example locally?

**Comment by random11x** [comment]
It does work locally

**Comment by random11x** [comment]
When using cli to set up the base project.

**Comment by seveves** [comment]
It seems like there is a problem on the angular material site:
https://github.com/angular/material2/issues/10086

**Comment by random11x** [comment]
while I havent tried adding in some local file. The comment at the bottom does say "You're right. For some reason, the date adapter is not being injected. Nothing to do with the locales file." So I dont think that would solve the problem anyway.

**Comment by EricSimons** [comment]
Hey all- just posted an update about this over in the material repo: https://github.com/angular/material2/issues/9780#issuecomment-375164307

tldr - I think this is due to differences in the UMD vs ESM modules material ships with. Need to investigate further and circle back. (This would mean t's not a bug with stackblitz but instead material itself, but we shall see :)

**Comment by purplem1lk** [comment]
Thank you for your patience - We are launching new core technology into beta this month that should resolve this issue. If this issue persists, feel free to tag me or reopen this issue with screenshots of the error. Thank you! :)
